index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/FetchInstitutionReportHandlerTest.java

```diff
@@ -24,103 +49,263 @@
+    private static SearchClient<NviCandidateIndexDocument> openSearchClient;
     private ByteArrayOutputStream output;
     private FetchInstitutionReportHandler handler;
 
     @BeforeEach
     public void setUp() {
         output = new ByteArrayOutputStream();
-        handler = new FetchInstitutionReportHandler();
+        openSearchClient = mock(OpenSearchClient.class);
+        handler = new FetchInstitutionReportHandler(openSearchClient);
     }
 
     @Test
     void shouldReturnUnauthorizedWhenUserDoesNotHaveSufficientAccessRight() throws IOException {
         var institutionId = randomUri();
-        var request = createRequest(institutionId, CURRENT_YEAR, AccessRight.MANAGE_DOI).build();
+        var customerId = randomUri();
+        var request = createRequest(institutionId, AccessRight.MANAGE_DOI, customerId,
+                                    Map.of(YEAR, String.valueOf(CURRENT_YEAR))).build();
         handler.handleRequest(request, output, CONTEXT);
         var response = GatewayResponse.fromOutputStream(output, Problem.class);
 
         assertThat(response.getStatusCode(), is(Matchers.equalTo(HttpURLConnection.HTTP_UNAUTHORIZED)));
     }
 
     @Test
-    void shouldReturnEmptyXlsxFile()
+    void shouldReturnBadRequestIfPathParamYearIsInvalid() throws IOException {
+        var request = requestWithInvalidYearParam();
+        handler.handleRequest(request, output, CONTEXT);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(Matchers.equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("listSupportedMediaTypes")
+    void shouldReturnCandidatesWithApprovalsBelongingToUsersTopLevelOrganization(String mediaType)
         throws IOException {
-        handler.handleRequest(validRequest(MICROSOFT_EXCEL.toString()), output, CONTEXT);
+        var topLevelCristinOrg = randomCristinOrgUri();
+        var candidatesInIndex = mockCandidatesInOpenSearch(topLevelCristinOrg);
+        var expected = getExpectedReport(candidatesInIndex, topLevelCristinOrg);
+
+        handler.handleRequest(requestWithMediaType(mediaType, topLevelCristinOrg), output, CONTEXT);
+
         var decodedResponse = Base64.getDecoder().decode(fromOutputStream(output, String.class).getBody());
         var actual = fromInputStream(new ByteArrayInputStream(decodedResponse));
-        var expected = new ExcelWorkbookGenerator(List.of("header"), List.of(List.of("value")));
         assertEquals(expected, actual);
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {
-        "application/vnd.ms-excel",
-        "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
-    })
-    void shouldReturnMediaTypeMicrosoftExcelWhenRequested(String mediaType) throws IOException {
-        var request = validRequest(mediaType);
+    @Test
+    void shouldPerformSearchForGivenInstitutionAndYear() throws IOException {
```

Comment:
I'm testing this specifically because `openSearchClient` is mocked in the tests.
We have separate integration tests that assert that `openSearchClient` find expected candidates for search parameters `year` and `topLevelCristinOrg`